### PR TITLE
[Cpp] SetMetrics method added

### PR DESCRIPTION
**Pull request by AliRKat**

### CODE REVIEW

**Comment by AliRKat** [comment]
[[C++, Doc] Add SetMetrics section](https://trello.com/c/5QVKdL0K/1688-c-doc-add-setmetrics-section)

File: cpp/next.md [hunk 1]
```diff
@@ -635,6 +635,21 @@ cly::Counlty.getInstance().setUserDetails(userdetail);
 </p>
 <pre><code class="cpp">cly::Countly.getInstance().setSalt("salt");</code></pre>
 <h1 id="h_01HABV267WW07MSKNVAY9PE9HT">Other Features and Notes</h1>
+<h2 id="h_01HWABRGA5S8N9CJHT286D29M8">Custom Metrics</h2>
```
Comment (turtledreams) [inline cpp/next.md:638]: seems like it does not follow the document structure we have:
**Other Features and Notes**
...
* Example Integrations
* SDK storage and Requests
  * Setting Event Queue Threshold
...
...
* Custom Metrics

File: cpp/next.md [hunk 2]
```diff
@@ -635,6 +635,21 @@ cly::Counlty.getInstance().setUserDetails(userdetail);
...
+<p>
+  Device Metrics information is sent to the server with every 'begin session' request.
+  It's possible to override this information and provide custom metrics. For such
+  cases, the SetMetrics() method can be used before starting the SDK. Device metrics
+  that are provided by SetMetrics() method are, _os, _os_version, _device, _resolution,
+  _carrier, and _app_version respectively. For further information about these
+  params, please refer to
+  <a href="https://support.count.ly/hc/en-us/articles/9290669873305-A-Deeper-Look-at-SDK-concepts#h_01HJ5PA5GMQSE8ATC3FJ6VAGP3" target="_blank" rel="noopener noreferrer">here</a>.
```
Comment (turtledreams) [inline cpp/next.md:646]: looks like this link will be better. now when I checked the other one did not have _device

File: cpp/next.md [hunk 3]
```diff
@@ -635,6 +635,21 @@ cly::Counlty.getInstance().setUserDetails(userdetail);
 </p>
 <pre><code class="cpp">cly::Countly.getInstance().setSalt("salt");</code></pre>
 <h1 id="h_01HABV267WW07MSKNVAY9PE9HT">Other Features and Notes</h1>
+<h2 id="h_01HWABRGA5S8N9CJHT286D29M8">Custom Metrics</h2>
+<p>
+  Device Metrics information is sent to the server with every 'begin session' request.
```
Comment (turtledreams) [inline cpp/next.md:640]: lets add we send them when asking for remote config too

File: cpp/next.md [hunk 4]
```diff
@@ -635,6 +635,21 @@ cly::Counlty.getInstance().setUserDetails(userdetail);
...
+  cases, the SetMetrics() method can be used before starting the SDK. Device metrics
+  that are provided by SetMetrics() method are, _os, _os_version, _device, _resolution,
+  _carrier, and _app_version respectively. For further information about these
+  params, please refer to
+  <a href="https://support.count.ly/hc/en-us/articles/9290669873305-A-Deeper-Look-at-SDK-concepts#h_01HJ5PA5GMQSE8ATC3FJ6VAGP3" target="_blank" rel="noopener noreferrer">here</a>.
+</p>
+<p>The example usage of SetMetrics() would be like this:</p>
+<pre><code class="cpp">Countly &amp;ct = Countly::getInstance();
+// OS, OS_version, device, resolution, carrier, app_version
+ct.SetMetrics("Windows 10", "10.22", "Mac", "800x600", "Carrier", "1.0");
```
Comment (turtledreams) [inline cpp/next.md:651]: 'mac' can be 'Lenovo'

**Review by turtledreams — CHANGES_REQUESTED** [review]

**Review by turtledreams — APPROVED** [review]

**Pull request merged**